### DISCUSSION TITLE
No trace information on AKS

### DISCUSSION THREAD

**Discussion by bardagadvergur** [open-telemetry/opentelemetry-dotnet#3209] [Q&A]
Hi ... I am completely lost as to what my issue can be so reaching out for help. I've added OpenTelemetry tracing to my .net core 6.0 application and in my local dev environment running in docker everything works as expected and traces are being exported. This is my tracing config:

`
       public static IServiceCollection AddTracing(this IServiceCollection services, string appName, IConfiguration configuration)
        {

            services.AddOpenTelemetryTracing(builder =>
            {
                builder
                    .AddAspNetCoreInstrumentation()
                    .AddMassTransitInstrumentation()
                    .AddSource(appName)
                    .AddSqlClientInstrumentation(opt => opt.SetDbStatementForText = true)
                    .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(appName))
                    .AddJaegerExporter(opts =>
                    {
                        opts.AgentHost = configuration.GetValue("Telemetry:Host", "jaeger");
                        opts.AgentPort = int.Parse(configuration.GetValue("Telemetry:Port", "6831"));
                        opts.Protocol = OpenTelemetry.Exporter.JaegerExportProtocol.UdpCompactThrift;
                        opts.ExportProcessorType = ExportProcessorType.Simple;
                    })
                    .AddConsoleExporter(opt =>
                    {
                        opt.Targets = OpenTelemetry.Exporter.ConsoleExporterOutputTargets.Console;
                    });
            });
            return services;
        }
    }
`
So locally I can see my traces in Jaeger as well as in the console. But when I deploy this application to my AKS kubernetes cluster no traces seem to be exported. I've validated the connection between tempo & my service using netcat & tcpdump so connectivity should not be an issue. I've also exposed the jaeger port of tempo in my cluster and have sent traces from my local environment into tempo running in the cluster so tempo is accepting traces just fine. After I was certain network connectivity was not an issue I added the console exporter and to my surprise no traces are visible in the logs on AKS.

These are the OpenTelemetry packages I am using:

`

		 "OpenTelemetry" Version="1.2.0-rc4"
		 
		 "OpenTelemetry.Exporter.Console" Version="1.2.0-rc4"
		 
		 "OpenTelemetry.Exporter.Jaeger" Version="1.2.0-rc4"
		 
		 "OpenTelemetry.Extensions.Hosting" Version="1.0.0-rc9.1"
		 
		 "OpenTelemetry.Instrumentation.AspNetCore" Version="1.0.0-rc9.1"
		 
		 "OpenTelemetry.Instrumentation.SqlClient" Version="1.0.0-rc9.1
		 
		 "OpenTelemetry.Instrumentation.Http" Version="1.0.0-rc9.1"
		 
		 "OpenTelemetry.Contrib.Instrumentation.MassTransit" Version="1.0.0-beta2"
`

**Comment by andhesky** [comment]
I had the same issue. It seems that you need to include .StartWithHost() as is shown in the [asp.net example](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/AspNetCore/Program.cs#L123)

This is really not obvious and I'm not sure what the usecase is where I would not want it started with the host so that really seems like it should be the default.

**Comment by sathoril** [comment]
Hello!

I'm having this same issue, I can see Trace Logging on my app locally, but not when I deploy to aks. I tried @andhesky solution, but apparently, `StartWithHost` option is not available on the version I'm using (1.4.0).

Does anyone know how to solve this issue?

**Comment by clcl0n** [comment]
Hi

i have same issue i'm using 1.8.0 version traces are generated on localhost but when i depoy to aks no traces are generated bud logs and metrics are just fine.

```
        services
            .AddOpenTelemetry()
            .ConfigureResource(resource => resource.AddService(Telemetry.ServiceName))
            .WithMetrics(metrics => metrics
                .AddAspNetCoreInstrumentation()
                .AddHttpClientInstrumentation()
                .AddOtlpExporter())
            .WithTracing(tracing => tracing
                .AddAspNetCoreInstrumentation()
                .AddEntityFrameworkCoreInstrumentation()
                .AddNpgsql()
                .AddHttpClientInstrumentation()
                .AddOtlpExporter());
```

**Reply by CrazyPear111** [reply]
Hi, try `SetSampler(new AlwaysOnSampler())`

**Comment by CrazyPear111** [comment]
Hi all 
For the latest version the following helped me:

```
 otel.WithTracing(tracing =>
{
    tracing
        .AddAspNetCoreInstrumentation()
        .SetSampler(new AlwaysOnSampler())    //this setup works in kubernetes
        .AddOtlpExporter(builder =>
        {
            builder.Endpoint = new(appSettings.OpenTelemetry.OtlpExporter.Endpoint);
            builder.Protocol = OtlpExportProtocol.Grpc;
        });
});
```

[Info about samples](https://opentelemetry.io/docs/concepts/sampling)
